### ISSUE TITLE
a merge name function

### ISSUE THREAD

**Issue by NicolasBelloy** [diging/tethne#72]
Is it possible to merge name depending on the formatting in the the imported file ?
ex : Doe, J = Doe, John

**Comment by erickpeirson** [comment]
Yes, we could do something like that. Right now, full-names and initial-names are stored separately. In v0.7 (coming very soon), it's something like:

``` python
>>> paper.authors_init
[('MCMURRICH', 'J P'),...]
>>> paper.authors_full
[('MCMURRICH', 'JAMES PLAYFAIR'),...]
```

Since WoS is inconsistent in whether it records include author full names, some papers have `authors_full` and others don't. The `authors` property will return the value of `authors_full` if it's available; otherwise, it returns `authors_init`.

``` python
>>> paper.authors
[('MCMURRICH', 'JAMES PLAYFAIR'),...]
>>> del paper.authors_full
>>> paper.authors    # authors_full is no longer available.
[('MCMURRICH', 'J P'),...]
```

So there are quite a few things that we could do to match full-name and initial-name representations, depending on the use-case. Here are some places we could do the matching:
- When we index a `Corpus` by authors;
- In network-building methods that evaluate author names (e.g. `coauthors`);
- ???

Could you say more about the use-cases that you have in mind? That might help us hone in on a good solution. Thanks for your help!
